src/types/redis_stream_base.h

```diff
@@ -239,6 +239,36 @@ struct StreamAutoClaimResult {
   std::vector<std::string> deleted_ids;
 };
 
+struct StreamPendingOptions {
+  uint64_t idle_time = 0;
+  bool with_time = false;
+
+  StreamEntryID start_id{StreamEntryID::Minimum()};
+  StreamEntryID end_id{StreamEntryID::Maximum()};
+
+  uint64_t count;
+  bool with_count = false;
+  bool with_consumer = false;
+
+  std::string consumer;
+  std::string stream_name;
+  std::string group_name;
+};
+
+struct StreamGetPendingEntryResult {
+  uint64_t pending_number;
+  StreamEntryID smallest_id;
```

Comment:
Do you think if it's a good idea to rename `smallest_id` and `greatest_id` to `first_entry_id` and `last_entry_id` to keep consistent with other places?

Reply:
I'll rename it.